framework/include/physics/PhysicsBase.h

```diff
@@ -168,13 +168,77 @@ class PhysicsBase : public Action, public InputParametersChecksUtils<PhysicsBase
    * @param blocks the vector blocks to check for whether it contains every block in the mesh
    */
   bool allMeshBlocks(const std::vector<SubdomainName> & blocks) const;
+  bool allMeshBlocks(const std::set<SubdomainName> & blocks) const;
 
   /**
    * Process the given petsc option pairs into the system solver settings
    */
   void addPetscPairsToPetscOptions(
       const std::vector<std::pair<MooseEnumItem, std::string>> & petsc_pair_options);
 
+  // Helpers to check on variable types
+  /// Whether the variable is a finite volume variable
+  bool isVariableFV(const VariableName & var_name) const;
+  /// Whether the variable is a scalar variable (global single scalar, not a field)
+  bool isVariableScalar(const VariableName & var_name) const;
+
+  // Routines to help with deciding when to create objects
+  /**
+   * Returns whether this Physics should create the variable or if it already exists
```

Comment:
Which of these ors is it? Assuming just the first
```suggestion
   * \returns whether this Physics should create the variable. Will return \p false if the variable already exists
```

Reply:
your wording. Let me add it